src/main/java/org/zalando/nakadi/domain/AdminAuthorization.java

```diff
@@ -0,0 +1,124 @@
+        // and then uses getters to check if values are null or not, so we need to do this check to avoid exception
+        this.admins = admins == null ? null : Collections.unmodifiableList(admins);
+        this.readers = readers == null ? null : Collections.unmodifiableList(readers);
+        this.writers = writers == null ? null : Collections.unmodifiableList(writers);
+    }
+
+    public List<AuthorizationAttribute> getAdmins() {
+        return admins;
+    }
+
+    public List<AuthorizationAttribute> getReaders() {
+        return readers;
+    }
+
+    public List<AuthorizationAttribute> getWriters() {
+        return writers;
+    }
+
+    public List<AuthorizationAttribute> getList(final AuthorizationService.Operation operation)
+            throws UnknownOperationException {
+        switch (operation) {
+            case ADMIN:
+                return admins;
+            case READ:
+                return readers;
+            case WRITE:
+                return writers;
+            default:
+                throw new UnknownOperationException("Unknown operation: " + operation.toString());
+        }
+    }
+
+    public List<Permission> toPermissionsList(final String resource) {
+        final List<Permission> permissions = admins.stream()
+                .map(p -> new Permission(resource, AuthorizationService.Operation.ADMIN, p))
+                .collect(Collectors.toList());
+        permissions.addAll(readers.stream()
+                .map(p -> new Permission(resource, AuthorizationService.Operation.READ, p))
+                .collect(Collectors.toList()));
+        permissions.addAll(writers.stream()
+                .map(p -> new Permission(resource, AuthorizationService.Operation.WRITE, p))
+                .collect(Collectors.toList()));
+        return permissions;
+    }
+
+    public static AdminAuthorization fromPermissionsList(final List<Permission> permissions) {
+        final List<AuthorizationAttribute> admins = permissions.stream()
+                .filter(p -> p.getOperation().equals(AuthorizationService.Operation.ADMIN))
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> readers = permissions.stream()
+                .filter(p -> p.getOperation().equals(AuthorizationService.Operation.READ))
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+        final List<AuthorizationAttribute> writers = permissions.stream()
+                .filter(p -> p.getOperation().equals(AuthorizationService.Operation.WRITE))
+                .map(p -> p.getAuthorizationAttribute())
+                .collect(Collectors.toList());
+
+        return new AdminAuthorization(admins, readers, writers);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
```

Comment:
Why do you define it? I do not see any usage of it, but I see that Permission class is used by Predicate.isEquals, which uses equals method, which is not defined for Permission.

Reply:
79d774e and fb90914